asllib/backend.mli

```diff
@@ -64,15 +65,17 @@ module type S = sig
       to compute the second operation. *)
 
   val bind_seq : 'a m -> ('a -> 'b m) -> 'b m
-  (** Monadic bind operation. but that only pass internal interpreter data.
+  (** Monadic bind operation, but that only passes internal interpreter data.
       This should not create any data-dependency. *)
 
   val bind_ctrl : 'a m -> ('a -> 'b m) -> 'b m
   (** Monadic bind operation, but that creates a control dependency between the
       first argument and the result of the second one. *)
 
-  val prod : 'a m -> 'b m -> ('a * 'b) m
-  (** Monadic product operation, two monads are combined "in parrallel".*)
+  val bind_par : 'a m -> 'b m -> ('a * 'b) m
+  (** Monadic product operation, two monads are combined "in parallel".*)
+
+  val appl: 'a m -> ('a -> 'b) -> 'b m
```

Comment:
I think it should be called `appl_data`, and that we should add a documentation comment:
```suggestion
  val appl_data: 'a m -> ('a -> 'b) -> 'b m
  (** Applicative map. 
  
      Creates a data dependency between the output events and the input events of the argument in the resulting monad. *)
```